db/job_context.h

```diff
@@ -83,45 +143,36 @@ struct JobContext {
   size_t num_alive_log_files = 0;
   uint64_t size_log_to_delete = 0;
 
-  explicit JobContext(int _job_id, bool create_superversion = false) {
+  explicit JobContext(int _job_id, bool create_superversion = false)
+    : superversion_context(create_superversion) {
     job_id = _job_id;
     manifest_file_number = 0;
     pending_manifest_file_number = 0;
     log_number = 0;
     prev_log_number = 0;
-    new_superversion = create_superversion ? new SuperVersion() : nullptr;
   }
 
   // For non-empty JobContext Clean() has to be called at least once before
   // before destruction (see asserts in ~JobContext()). Should be called with
   // unlocked DB mutex. Destructor doesn't call Clean() to avoid accidentally
   // doing potentially slow Clean() with locked DB mutex.
```

Comment:
I mean in a future refactoring :)

Reply:
I refactored the code so that all the SuperVersions are always deleted outside of the db_mutex using a sv_context.
So yes, it should be ok to call Clean() from the destructor, but I still agree with the comment: the cleanup is a heavy operation and it should be explicit and not 'hidden' in the destructor.